web3.js/src/transaction.ts

```diff
@@ -327,17 +327,24 @@ export class Transaction {
       return this._message;
     }
 
-    const {nonceInfo} = this;
-    if (nonceInfo && this.instructions[0] != nonceInfo.nonceInstruction) {
```

Comment:
Bug here. If the nonce advance instruction _already happened_ to be at the head of the list (eg. because someone put it there) we wouldn't end up using the nonce as the `recentBlockhash` in the `Message`.